patches/@endo+patterns+0.2.2.patch

```diff
@@ -0,0 +1,31 @@
+diff --git a/node_modules/@endo/patterns/src/types.d.ts b/node_modules/@endo/patterns/src/types.d.ts
+index bc09e84..ff2d21f 100644
+--- a/node_modules/@endo/patterns/src/types.d.ts
++++ b/node_modules/@endo/patterns/src/types.d.ts
+@@ -485,7 +485,10 @@ export type InterfaceGuard = {
+  * }
+  * ```
+  */
+-export type MethodGuardMaker = any;
++export type MethodGuardMaker = {
++    optional: (...argGuards: ArgGuard[]) => MethodGuardMaker,
++    returns: (r?: Pattern) => MethodGuard,
++};
+ export type MethodGuard = {
+     klass: 'methodGuard';
+     callKind: 'sync' | 'async';
+diff --git a/node_modules/@endo/patterns/src/types.js b/node_modules/@endo/patterns/src/types.js
+index 7a4704a..caf9382 100644
+--- a/node_modules/@endo/patterns/src/types.js
++++ b/node_modules/@endo/patterns/src/types.js
+@@ -509,7 +509,9 @@ export {};
+  */
+ 
+ /**
+- * @typedef {any} MethodGuardMaker
++ * @typedef {{
++ *   optional: (...argGuards: ArgGuard[]) => MethodGuardMaker,
++ * }} MethodGuardMaker
```

Comment:
There's also a `rest` method, and a required method invocation order (here and in the above types.d.ts changes):
```suggestion
+ * @typedef {{
+ *   optional: (...argGuards: ArgGuard[]) => Omit<MethodGuardMaker, 'optional'>,
+ *   rest: (argGuard: ArgGuard) => Omit<MethodGuardMaker, 'optional' | 'rest'>,
+ *   returns: (returnGuard?: Pattern) => MethodGuard,
+ * }} MethodGuardMaker
```

This also needs an Endo issue/PR.

Reply:
gah, this I had them all in there I don't know what happened. I'll amend.

To get into Endo I expect Endo patches in SDK to get migrated there routinely.

Reply:
Reverted as out of scope